tests/test_embit_utils.py

```diff
@@ -369,8 +369,14 @@ def test_parse_derivation_path():
         (SC.TESTNET, SC.CUSTOM_DERIVATION, True): "m/45'/1'/0'/1/5",
         (SC.REGTEST, SC.CUSTOM_DERIVATION, True): "m/45'/1'/0'/1/5",
 
-        # CRAZY custom derivation path
-        (None, SC.CUSTOM_DERIVATION, False): "m/879345978543'/908327034508534983495'/9085098430894380959043'/0/5",
+        # CRAZY custom derivation paths
+        (None, SC.CUSTOM_DERIVATION, False, 5): "m/879345978543'/908327034508534983495'/9085098430894380959043'/0/5",
```

Comment:
I believe the first 3 nodes of this derivation path are invalid because max hardened representation is 2**31-1, which is really representing the max child node 2**32-1.  If we were to pass these thru to embit, it would raise an exception eventually because each child needs to fit into a 4byte serialization.

Reply:
maybe all tests above can also check for the 'index' element on return, if I'm understanding correctly that it's always there.

Reply:
Seemed unnecessarily fussy / repetitive to verify index in the base cases, but was a late addition here to verify the new index parsing logic in this PR.

Even with just three components in the tuple it's hard to keep straight when reading the code. The few 4-tuples added at the end make it even harder. Not super-concerned since it's not in the main app code, but an easier to follow refactor would eventually be nice.

Reply:
Slightly more realistic _intentionally unrealistic_ test derivation path updated.